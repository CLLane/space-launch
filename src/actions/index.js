export const nextLaunchData = (launch) => ({
  type: 'NEXT_LAUNCH',
  launch,
})

export const monthLaunchData = (launches) => ({
  type: 'MONTH_LAUNCH',
  launches
})

export const rocketTypeData = (rockets) => ({
  type: 'ROCKET_TYPE',
  rockets
})

export const getMissionData = (missions) => ({
  type: 'MISSION_DATA',
  missions
})

export const getCompanyData = (companies) => ({
  type: 'COMPANY_DATA',
  companies
})