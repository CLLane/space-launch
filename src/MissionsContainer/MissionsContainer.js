import React from 'react';
import { Component } from 'react';
import { getMissions} from '../apiCalls/apiCalls';
import { MissionCard } from '../Cards/MissionCards';
import { connect } from 'react-redux';
import { getMissionData } from '../actions';

export class MissionsContainer extends Component {

  async componentDidMount() {
    let missionType = await getMissions();
    this.props.getMissionData(missionType)
  }
  
  render() {
    let missions = this.props.missions.map(mission => <MissionCard mission={mission} />)
    return (
      <>
        {missions}
      </>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  getMissionData: (missions) => dispatch(getMissionData(missions))
})

export const mapStateToProps = ({ missions }) => ({
  missions
})

export default connect(mapStateToProps, mapDispatchToProps)(MissionsContainer)
