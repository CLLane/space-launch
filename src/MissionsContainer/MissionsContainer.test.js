import React from 'react';
import { shallow } from "enzyme";
import { MissionsContainer, mapDispatchToProps, mapStateToProps } from "../MissionsContainer/MissionsContainer";
import { getMissionData } from "../actions/index";

describe('MissionsCOntainer', () => {
  let wrapper;
  let mockGetMissionData = jest.fn();
  let mockMissions = [{name: 'Operation get to mod 4'}]
  let mockMission = {
    name: 'Operation get to mod 4'
  };

  beforeEach(() => {
    wrapper = shallow(
      <MissionsContainer 
        mission={mockMission}
        getMissionData={mockGetMissionData}
        missions={mockMissions}
      />
    );
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return an object with a missions array', () => {
    const mockState = {
      missions: [{name: 'Apollo'}, {name: 'Falcon'}]
    }
    const expected = {
      missions: [{ name: 'Apollo' }, { name: 'Falcon' }]
    }
    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected)
  });
});

describe('mapDispatchToProps', () => {
  it('calls dispatch with a getMissionData action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = getMissionData(['mock'])
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.getMissionData(['mock'])

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})