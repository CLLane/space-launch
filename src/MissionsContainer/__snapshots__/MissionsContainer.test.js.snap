// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsCOntainer should match a snapshot 1`] = `
<Fragment>
  <MissionCard
    mission={
      Object {
        "name": "Operation get to mod 4",
      }
    }
  />
</Fragment>
`;
